R/utils.R

```diff
@@ -218,3 +218,37 @@
   return(invisible(NULL))
 }
 
+#' Utility function to capture warnings and errors
+#'
+#' @param expr Expression to evaluate
+#'
+#' @noRd
+evaluate_with_catch <- function(expr) {
+  result <- NULL
+  warning_message <- NULL
+  error_message <- NULL
+
+  tryCatch(
+    {
+      # Capture warnings
+      withCallingHandlers(
+        {
+          result <- eval(expr)
+        },
+        warning = function(w) {
+          warning_message <<- w$message
```

Comment:
Should this be saved globally to `warning_message`?

Reply:
Here's a note from the documentation of `<<-`:
_The operators <<- and ->> are normally only used in functions, and cause a search to be made through parent environments for an existing definition of the variable being assigned. If such a variable is found (and its binding is not locked) then its value is redefined, otherwise assignment takes place in the global environment. _

Since these variables (`warning_message` and `error_message`) are defined in the parent environment, `<<-` will update these variables in the parent environment.